### DISCUSSION TITLE
Python dictionary highlighter

### DISCUSSION THREAD

**Discussion by TheMessik** [dodona-edu/feedback-and-support#18] [Feature requests]
When comparing dictionaries in tests, students are often presented with a visual similar to one below when they get a test case wrong.
![image](https://user-images.githubusercontent.com/54912465/233099363-5f9099c5-ab71-43f8-a8a8-e0f969e80f0e.png)

This makes debugging their code extremely tedious.

Requested feature: highlight the differences in the expected vs actual dictionary, like it's done when comparing strings.

**Comment by bmesuere** [comment]
The reason why your screenshot isn't optimal is twofold:

We normally do highlight differences within the line to make it easier to spot the mistakes. For exercises with a lot of output/tests it can take a while (sometimes up to a minute) to generate those diffs and it produces a lot of html causing some browser to crash. This is why we heuristically fall back to a simpler diffing algorithm. In this case, the long line length in your screenshot triggered our fallback.

Even if we would use the complete diff, it still wouldn't be easy to spot the mistakes in this case. This is because the order of elements in dictionaries isn't guaranteed. The creator of the exercise (@pdawyndt ?) could possibly improve this by tweaking the exercise or tests.
